### PR TITLE
Storybook version bump and clean up

### PR DESCRIPTION
**Pull request by alex-page**
- [x] Reverts Shopify/polaris#6791
- [x] Bumps storybook versions
- [x] Cleans up `main.js` in storybook
- [x] Fixes issue with multiple versions of React and react types that were causing issues building the website

### CODE REVIEW

File: .changeset/green-trees-beam.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Bump storybook versions and clean up main.js
```
Comment (chloerice) [inline .changeset/green-trees-beam.md]:
```suggestion
Bumped storybook versions and cleaned up main.js
```

**Review by chloerice — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
Can successfully run `yarn && yarn build` and storybook/styleguide locally 🚀 💯

File: polaris-for-figma/package.json [hunk 2]
```diff
@@ -15,14 +15,14 @@
...
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
     "sass": "^1.32.8"
   },
   "devDependencies": {
     "@figma/plugin-typings": "^1.19.1",
-    "@types/react": "^16.0.0",
-    "@types/react-dom": "^16.0.0",
+    "@types/react": "*",
+    "@types/react-dom": "*",
```
Comment (alex-page) [inline polaris-for-figma/package.json:25]:
```suggestion
    "@types/react": "^17.0.2",
    "@types/react-dom": "^17.0.2",
```

Comment (alex-page) [inline polaris-for-figma/package.json:25]:
```suggestion
    "@types/react": "^17.0.2",
    "@types/react-dom": "^17.0.2",
```

File: polaris.shopify.com/package.json [hunk 3]
```diff
@@ -41,7 +41,7 @@
     "@types/marked": "^4.0.3",
     "@types/node": "17.0.21",
     "@types/prismjs": "^1.26.0",
-    "@types/react": "17.0.39",
+    "@types/react": "*",
```
Comment (alex-page) [inline polaris.shopify.com/package.json:44]: The site cannot resolve the types without the `"*"`

**Review by sam-b-rose — APPROVED** [review]

**Pull request merged**